### DISCUSSION TITLE
How sqlmodel implements the same functionality as SerializerMethodField in DRF?

### DISCUSSION THREAD

**Discussion by rubbish822** [fastapi/sqlmodel#786] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from rest_framework import serializers
from drf_spectacular.utils import extend_schema_field


class UserSerializer(serializers.ModelSerializer):
    full_name = serializers.SerializerMethodField(read_only=True)
    
    @extend_schema_field(serializers.CharField)
    def full_name(self, obj):
        return f'{obj.first_name}:{obj.id}'
```
```


### Description

```
from rest_framework import serializers
from drf_spectacular.utils import extend_schema_field

class UserSerializer(serializers.ModelSerializer):
    full_name = serializers.SerializerMethodField(read_only=True)
    
    @extend_schema_field(serializers.CharField)
    def full_name(self, obj):
        return f'{obj.first_name}:{obj.id}'
```

### Operating System

macOS

### Operating System Details

_No response_

### SQLModel Version

0.0.14

### Python Version

Python 3.11.1

### Additional Context

_No response_

**Comment by YuriiMotov** [comment, marked as answer]
In SQLMOdel you would use `computed_field`:

```py
from pydantic import computed_field
from sqlmodel import Field, Session, SQLModel, create_engine


class UserBase(SQLModel):
    id: int | None = Field(default=None, primary_key=True)
    first_name: str

class User(UserBase, table=True):
    pass

class UserRead(UserBase):
    @computed_field
    @property
    def full_name(self) -> str:
        return f"{self.first_name}:{self.id}"


sqlite_url = "sqlite:///"

engine = create_engine(sqlite_url)
SQLModel.metadata.create_all(engine)


with Session(engine) as session:
    user = User(first_name="John")
    session.add(user)
    session.commit()

    user_read = UserRead.model_validate(user)
    assert user_read.full_name == "John:1"

```
